src/pkg/credentials/aws.go

```diff
@@ -22,13 +20,18 @@ type AWS struct {
 
 // GetAWS is a helper for aggregating aws secrets and credentials.
 func GetAWS(override map[string]string) AWS {
-	accessKey := os.Getenv(AWSAccessKeyID)
+	var accessKey, secretKey, sessToken string
 	if ovr, ok := override[AWSAccessKeyID]; ok && ovr != "" {
 		accessKey = ovr
 	}
 
-	secretKey := os.Getenv(AWSSecretAccessKey)
-	sessToken := os.Getenv(AWSSessionToken)
+	if ovr, ok := override[AWSSecretAccessKey]; ok && ovr != "" {
+		secretKey = ovr
+	}
```

Comment:
the `ok` check on these is unnecessary.  Either they're going to get replaced by a value, or they're going to remain as empty strings.  
```suggestion
	secretKey := override[AWSSecretAccessKey]
```